reporters/reporter.rb

```diff
@@ -0,0 +1,55 @@
+require_relative '../registry'
+
+module Reporters
+  class Reporter
+    @@global_registry = Registry::MetricsRegistry.new
+
+    attr_reader :registry
+
+    def initialize(registry = nil, reporting_interval = 5)
+      @registry = registry || @@global_registry
+      @reporting_interval = reporting_interval
+      @lock = Mutex.new
+      @closed = true
+      @task = nil
+    end
+
+    # Start reporting
+    def start
+      @lock.synchronize do
+        if @closed
+          @closed = false
+          @task = Thread.start {schedule_task}
+        end
+      end
+    end
+
+    # Stop reporting
+    def stop
+      # Flush all buffer before close the client.
+      @lock.synchronize do
+        @closed = true
+        @task.join

```

Comment:
@yogeshprasad While this statement is logically correct (and I understand I suggested this approach), this could cause a lot of waiting when `reporting_interval` is set to a large value. I realized this when I was writing code for  heartbeat service and I think I have a better approach. I'll share it tomorrow. Apart from that, this looks fine.

The latest commit also includes some changes needed for my work.